packages/embeds/embed-core/playwright/tests/embed-pages.e2e.ts

```diff
@@ -45,18 +47,77 @@ test.describe("Embed Pages", () => {
     expect(marginFromTop).not.toBe(0);
   });
 
-  test("should change to embed when window.name is changed to cal-embed=", async ({ page }) => {
-    await page.goto("http://localhost:3000/free/30min");
+  test.describe("isEmbed, getEmbedNamespace, getEmbedTheme testing", () => {
+    test("when `window.name` is set to 'cal-embed=' and `theme` is supplied as a query param", async ({
+      page,
+    }) => {
+      const queryParamTheme = "dark";
+      await page.evaluate(() => {
```

Comment:
We can set window.name before going to the URL and then load the URL. It avoids an unnecessary page load saving time per test.